lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_correct_edge_insets_constructor/validator.dart

```diff
@@ -0,0 +1,191 @@
+part of 'prefer_correct_edge_insets_constructor_rule.dart';
+
+class _Validator {
+  final _exceptions = <InstanceCreationExpression, String>{};
+
+  _Validator();
+
+  Map<InstanceCreationExpression, String> get expressions => _exceptions;
+
+  void validate(Map<InstanceCreationExpression, EdgeInsetsData> expressions) {
+    for (final expression in expressions.entries) {
+      final data = expression.value;
+
+      if (data.constructorName == _constructorNameFromSTEB ||
+          data.constructorName == _constructorNameFromLTRB) {
+        _validateFromSTEB(expression.key, data);
+      }
+
+      if (data.constructorName == _constructorNameSymmetric) {
+        _validateSymmetric(expression.key, data);
+      }
+      if (data.constructorName == _constructorNameOnly) {
+        _validateFromOnly(expression.key, data);
+      }
+    }
+  }
+
+  void _validateSymmetric(InstanceCreationExpression key, EdgeInsetsData data) {
+    final param = data.params;
+    final vertical = param.firstWhereOrNull((e) => e.name == 'vertical')?.value;
+    final horizontal =
+        param.firstWhereOrNull((e) => e.name == 'horizontal')?.value;
+
+    final isParamsSame = vertical == horizontal && horizontal != null;
+    final isAllParamsZero = isParamsSame && horizontal == 0;
+
+    if (isAllParamsZero) {
+      _exceptions[key] = _replaceWithZero();
+    }
+
+    if (isParamsSame) {
+      _exceptions[key] = '${data.className}.all($horizontal)';
+    }
+
+    if (horizontal == 0 && vertical != null) {
+      _exceptions[key] = _replaceWithSymmetric('vertical: $vertical');
+    }
+    if (vertical == 0 && horizontal != null) {
+      _exceptions[key] = _replaceWithSymmetric('horizontal: $horizontal');
+    }
```

Comment:
I would suggest to return expected replacement instead of assignment and assign it in `validate`